service/storage_service.cc

```diff
@@ -1485,6 +1485,11 @@ future<> storage_service::gossip_sharder() {
 }
 
 future<> storage_service::stop() {
```

Comment:
This method is not called in scylla, only in `cql_test_env`.

Reply:
Wow. Not sure what to do. The series will work the same way as the current code (which abandons the future), but not stopping stuff is not good.

Reply:
> Not sure what to do. 

I think we should add 
```c++
auto stop_storage_service = defer_verbose_shutdown([&ss] {
    ss.invoke_on_all(&service::storage_service::stop).get(); 
});
```
right after `ss.start(...)` in `main()`. That's almost the code that will eventually be there.

> The series will work the same way as the current code (which abandons the future),

Why the same? Before the patch the call to `maybe_rewrite_streams_descriptions` was waited on inside `join_token_ring`. Adter the patch it's sent to the background and effectively becomes abandoned.

Reply:
> Why the same? Before the patch the call to maybe_rewrite_streams_descriptions was waited on inside join_token_ring. Adter the patch it's sent to the background and effectively becomes abandoned.

The first part of `maybe_rewrite_streams_descriptions` was indeed waited on, then there was a second part which was sent into the background. Now the entire function is sent into background. But that's also fine, nothing requires the stuff in the first part to be finished before we proceed (the first part may only affect results of other executions of this function on this node which only happen once during Scylla startup).

Reply:
But the things maybe_rewrite_streams_description require that the system not be deallocated under their feet.

Reply:
(so I agree with @xemul and disagree with myself)